core-container/entry.sh

```diff
@@ -242,51 +230,68 @@ function run_kcp() {
 
 function run_init() {
     echo "--< Starting init >--"
-    wait_kcp_ready
     set_provider_adapters
-    kubestellar init --local-kcp false --ensure-imw $ENSURE_IMW --ensure-wmw $ENSURE_WMW
-    kubectl ws root
+    kubestellar init --ensure-imw $ENSURE_IMW --ensure-wmw $ENSURE_WMW 
     touch ready
     echo "***READY***"
     sleep infinity
 }
 
-
 function run_mailbox_controller() {
     echo "--< Starting mailbox-controller >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! mailbox-controller -v=${VERBOSITY} ; then
         echoerr "unable to start mailbox-controller!"
         exit 1
     fi
 }
 
-
 function run_where_resolver() {
     echo "--< Starting where-resolver >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! kubestellar-where-resolver -v ${VERBOSITY} ; then
         echoerr "unable to start kubestellar-where-resolver!"
         exit 1
     fi
 }
 
-
 function run_placement_translator() {
     echo "--< Starting placement-translator >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! placement-translator --allclusters-context  "system:admin" -v=${VERBOSITY} ; then
         echoerr "unable to start mailbox-controller!"
         exit 1
     fi
 }
 
+# set_host_kubeconfig: 
```

Comment:
The name "set_host_kubeconfig" is odd for this function. That name suggests that some whole-file sort of setting is being done, such as setting somewhere (where?) an indication of _which_ kubeconfig to use. But what this function actually does is modify an existing kubeconfig to ensure the existence and content of a particular cluster, user, and context and set the current-context to refer to that context. Perhaps a better name would be something like "set_space_mgt_context"?

Reply:
I changes it to set_kubestellar_kubeconfig.  The purpose of this function is to get us the kubeconfig used by kubestellar.  We are currently using this config for checking that certain pods are "ready" (and yes this is a bad way to do it and we should change it, but it's currently working in production and we'll get it to it in the future).  It is also being used by some kcp specific logic and there we can also change it to use the kcp config stored in the secret instead.